### PR TITLE
Remove images from Docker Hub from tests

### LINKED ISSUE #3498: Docker Hub: 429s keep breaking the builds
### Description

Docker Hub has stringent (non-sensical?) rate limitations on their service that no other registry provider impose.

As such, Hub will rapidly respond 429 - effectively breaking our tests.
Since this "temporary" ban apply for some time, it will further prevent any re-attempt.

Suggesting we either "pre-heat" all needed images at the start of our test suites, so that at least if pull are going to fail, we fail fast - or we remove all of these images from our test suite and use copies stored on ghcr.

### Steps to reproduce the issue

https://github.com/containerd/nerdctl/actions/runs/11188098475/job/31106292438?pr=3497#step:7:524

### Describe the results you received and expected

429

### What version of nerdctl are you using?

main

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information

_No response_

### PR DESCRIPTION
**Pull request by apostasie**
Fix #3498

### CODE REVIEW

**Comment by apostasie** [comment]
Note that this here is highlighting another of our test issues: we do have a number of platform specific files, but we ALSO do mark certain tests as skipped if we are on windows.

Using the two mechanisms simultaneously is problematic, and leading to situations like this here, where we have to add non-sensical definitions.

Suggestion is to stop using platform specific files for tests entirely, and instead just use appropriate `Skip`.
- easier to know how many / which tests are being skipped on a certain platform
- easier to enable / disable a single test for a platform without having to move a bunch of files around

**Comment by apostasie** [comment]
TestRunWithTtyAndDetached #3437 

Besides that ^, it is ready.

@AkihiroSuda at your convenience.

File: cmd/nerdctl/image/image_list_test.go [hunk 1]
```diff
@@ -96,10 +96,9 @@ LABEL version=0.1`, testutil.CommonImage)
 	defer base.Cmd("rmi", tempName).AssertOK()
 
 	busyboxGlibc, busyboxUclibc := "busybox:glibc", "busybox:uclibc"
-	base.Cmd("pull", busyboxGlibc).AssertOK()
+	base.Cmd("tag", testutil.CommonImage, busyboxGlibc).Run()
+	base.Cmd("tag", testutil.CommonImage, busyboxUclibc).Run()
```
Comment (AkihiroSuda) [inline cmd/nerdctl/image/image_list_test.go:100]: Needs a comment line to explain why not pull real images

Comment (AkihiroSuda) [reply cmd/nerdctl/image/image_list_test.go:100]: I also wonder if we can avoid using fake "busybox" names

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

File: cmd/nerdctl/image/image_list_test.go (hunk 1 above)
Comment (apostasie) [reply cmd/nerdctl/image/image_list_test.go:100]: Yeah - I'll just use some other name instead - it doesn't really matter.

Comment (apostasie) [reply cmd/nerdctl/image/image_list_test.go:100]: > Needs a comment line to explain why not pull real images

Well, nothing is really "real" in our software world, right? :-) - jk :-)

I will write a comment.

**Comment by fahedouch** [comment]
Hi @apostasie, there is also a rate limiting  on ghcr and everywhere!
A non-expensive solutions would be :
- to check if image exist in content storage before making a pull otherwise pull 
- cache image list in mem struct for each pull and make a mem check for each pull.

unfortunately there is no flag that does this check in the pull subcommand

**Comment by apostasie** [comment]
> Hi @apostasie, there is also a rate limiting on ghcr and everywhere! A non-expensive solutions would be :
> 
> * to check if image exist in content storage before making a pull otherwise pull
> * cache image list in mem struct for each pull and make a mem check for each pull.
> 
> unfortunately there is no flag that does this check in the pull subcommand

At the start of tests, we could spin a local registry and push there the images we need.

Then we could leverage hosts.toml to use the local registry as cache during the tests.

**Comment by apostasie** [comment]
@AkihiroSuda updated to remove the fake "busybox" name and added a comment.

**Comment by apostasie** [comment]
`TestRunWithTtyAndDetached`

**Comment by apostasie** [comment]
> Hi @apostasie, there is also a rate limiting on ghcr and everywhere! A non-expensive solutions would be :

I am sure there is - we just never hit it on the CI, or debugging locally...

File: cmd/nerdctl/image/image_list_test.go [hunk 2]
```diff
@@ -95,12 +95,12 @@ LABEL version=0.1`, testutil.CommonImage)
...
-	defer base.Cmd("rmi", busyboxGlibc).AssertOK()
-
-	base.Cmd("pull", busyboxUclibc).AssertOK()
-	defer base.Cmd("rmi", busyboxUclibc).AssertOK()
+	// This test is about testing local filtering of image names - as such, we do not need remote images at all
+	taggedGlibc, taggedUclibc := "taggedimage:glibc", "taggedimage:uclibc"
+	base.Cmd("tag", testutil.CommonImage, taggedGlibc).Run()
+	base.Cmd("tag", testutil.CommonImage, taggedUclibc).Run()
+	defer base.Cmd("rmi", taggedGlibc).AssertOK()
+	defer base.Cmd("rmi", taggedUclibc).AssertOK()
```
Comment (AkihiroSuda) [inline cmd/nerdctl/image/image_list_test.go:103]: Why not just use simple meta syntax vars such as "foo" "bar"?
The implementation of libc doesn't really relate to the test.

Comment (apostasie) [reply cmd/nerdctl/image/image_list_test.go:103]: The existing test was using that - I just did not change it - and I don't think it matters one way or the other.
And yes, it is pretty much `vfoov` and `zzfoozz`

I am also not particularly interested in spending more energy than needed on these tests in their current form. They are neither particularly good nor complete enough, and I would rather focus on the rewrite.

Anyhow, I will update if you prefer.

Comment (apostasie) [reply cmd/nerdctl/image/image_list_test.go]: Done

Comment (apostasie) [reply cmd/nerdctl/image/image_list_test.go]: > The existing test was using that - I just did not change it - and I don't think it matters one way or the other. And yes, it is pretty much `vfoov` and `zzfoozz`
> 
> I am also not particularly interested in spending more energy than needed on these tests in their current form. They are neither particularly good nor complete enough, and I would rather focus on the rewrite.
> 
> Anyhow, I will update if you prefer.

Hope this didn't sound defensive.

I guess lately I have been working on tests / QA way too much for my own good :-).

Cheers.

**Comment by apostasie** [comment]
Latest failures are the usual - `TestIPFSComposeUp` and `TestRunWithTtyAndDetached `

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**